Modules/vc-module-catalog/VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleProductsController.cs

```diff
@@ -138,10 +138,14 @@ public async Task<ActionResult<CatalogProduct>> GetNewProductByCatalogAndCategor
             {
                 result.TryInheritFrom(parent);
             }
-            foreach (var property in result.Properties)
+
+            if (result.Properties.Any())
```

Comment:
If `result.Properties` is empty then there is no difference between `result.Properties.Any()` and `foreach`.

Reply:
It's not empty, it's null. So we get null reference